### PR TITLE
Remove finance dependencies in node and client/jackson and make finance a CorDapp.

### PR DESCRIPTION
**Pull request by mnesbit**
This PR removes the finance dependency fully from :node and the :client/jackson projects. All finance flows and schemas are now registered in standard CorDapp fashion.

To make this work I have extended the web plugin registry to allow custom JSON serializer registrations.

Also, finance needs to be referred to from build.gradle of sample CorDapps as compile not cordaCompile. For the minute this causes a warning, which Clinton and I will look into fixing.

### CODE REVIEW

**Comment by mnesbit** [comment]
The :finance module is now a full CorDapp now that a bug fix has been put into the plugins.

File: webserver/src/main/kotlin/net/corda/webserver/internal/NodeWebServer.kt [hunk 1]
```diff
@@ -127,8 +128,14 @@ class NodeWebServer(val config: WebServerConfig) {
             addServlet(DataUploadServlet::class.java, "/upload/*")
             addServlet(AttachmentDownloadServlet::class.java, "/attachments/*")
 
+
```
Comment (rick-r3) [inline webserver/src/main/kotlin/net/corda/webserver/internal/NodeWebServer.kt]: Extra blank line to remove.

File: finance/src/main/kotlin/net/corda/plugin/FinanceJSONSupport.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+fun registerFinanceJSONMappers(objectMapper: ObjectMapper): Unit {
+    val financeModule = SimpleModule("finance").apply {
+        addSerializer(BusinessCalendar::class.java, CalendarSerializer)
+        addDeserializer(BusinessCalendar::class.java, CalendarDeserializer)
+    }
+    objectMapper.registerModule(financeModule)
+}
+
```
Comment (rick-r3) [inline finance/src/main/kotlin/net/corda/plugin/FinanceJSONSupport.kt]: Extra blank line

File: finance/src/main/kotlin/net/corda/plugin/FinanceJSONSupport.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.core.JsonParseException
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.*
+import com.fasterxml.jackson.databind.deser.std.StringArrayDeserializer
+import com.fasterxml.jackson.databind.module.SimpleModule
+import net.corda.contracts.BusinessCalendar
+import java.time.LocalDate
+
+
```
Comment (rick-r3) [inline finance/src/main/kotlin/net/corda/plugin/FinanceJSONSupport.kt]: Extra blank line

**Pull request merged**